CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+
+### Code style
+
+Serverpod uses the Dart linter to enforce a consistent code style. The linter is run as part of the CI checks, so it is important that the code follows the linter rules. When you write code, make sure to use `dart format` and `dart analyze` to ensure that the code follows the linter rules.
+
+We try to follow the [Effective Dart](https://dart.dev/guides/language/effective-dart) guidelines as much as possible. But above all we care about code readability and maintainability. Therefore, we encourage you to write code that is easy to read and understand for future contributors.
+
+### Running the tests
+
+Serverpod has a comprehensive test suite that covers the core functionality of the project. The tests are run as part of the CI checks, so to speed up development it can be good to run the tests locally before submitting a pull request.
+
+To ensure all tests work as expected, it is recommended to add an entry for the test server, postgres and redis at the end of your `/etc/hosts file`.
+```text
+127.0.0.1 serverpod_test_server
+127.0.0.1 postgres
+127.0.0.1 redis
+```
+
+#### Test scripts
+
+Scripts that run groups of tests are located in the `util` directory and their name start with `run_tests`. The following test scripts are available:
+
+| Script | Description |
+| --- | --- |
+| `run_tests_unit` | Run all unit tests. |
+| `run_tests_integration` | Run all non concurrent integration tests in the test project. |
+| `run_tests_integration_concurrently` | Run all concurrent integration tests in the test project. |
+| `run_tests_flutter_integration` | Run all Flutter integration tests in the flutter test project. |
+| `run_tests_e2e` | Run all server end to end tests in the test project. |
+| `run_tests_migrations_e2e` | Run all migration end to end tests. |
+| `run_tests_bootstrap` | Run all bootstrap tests. |
+| `run_tests_update_pubspecs` | Ensure that all pubspec files are up to date with the templates. |
+| `run_tests_analyze` | Run the code analysis tests. |
+| `run_tests_update_pubspecs` | Ensure that all pubspec files are up to date with the templates. |
+
+To run any script, navigate to the root of the repository and run the script, e.g.:
+
+```bash
+$ util/run_tests_unit
+```
+
+#### Running single tests
+
+All unit tests can be run with the `test` command. But for some tests, you may need to perform additional steps to set up a test environment. Listed below are the tests that require additional setup.
+
+##### Running integration tests
+
+To run any integration test in the `tests/serverpod_test_server/test_integration` directory you will need to follow these steps:
+
+1. Start the Docker container for the test server.
+
+    ```bash
+    $ cd tests/serverpod_test_server/docker-local
+    $ docker-compose up --build --detach
```

Comment:
For my "Docker Desktop" installation I had to use `docker compose` now.


Seems like they dropped the standalone "docker-compose" a while back: https://docs.docker.com/compose/releases/migrate/#what-are-the-differences-between-compose-v1-and-compose-v2


With that change the guide worked smoothly :)